ui/src/loading.tsx

```diff
@@ -1,26 +1,65 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
 import CircularProgress from "@mui/material/CircularProgress";
-import Grid from "@mui/material/Grid";
-import React, { useEffect, useRef, useState } from "react";
+import Typography from "@mui/material/Typography";
+import React, { ReactNode, useEffect, useRef, useState } from "react";
 
-export default function Loading() {
+type State = {
```

Comment:
opt: I don't know if it's necessary to split out this definition into its own type. I haven't done too much experimentation in this area, but pretty sure you can just inline the object type into Props below, if you preferred that.

Reply:
I'm using it below for showStatus.